impl_core/src/test/java/io/opencensus/implcore/trace/propagation/B3SingleFormatTest.java

```diff
@@ -0,0 +1,284 @@
+  }
+
+  @Test
+  public final void verifyExtractParsesSampledRootSpanFromB3() throws SpanContextParseException {
+    // given
+    final Map<String, String> carrier =
+        ImmutableMap.of("b3", "4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-1");
+
+    // when
+    final SpanContext result = format.extract(carrier, getter);
+
+    // then
+    assertEquals("4bf92f3577b34da6a3ce929d0e0e4736", result.getTraceId().toLowerBase16());
+    assertEquals("00f067aa0ba902b7", result.getSpanId().toLowerBase16());
+    assertTrue(result.getTraceOptions().isSampled());
+  }
+
+  @Test
+  public final void verifyExtractParsesSampledRootSpanFromTracestate()
+      throws SpanContextParseException {
+    // given
+    final Map<String, String> carrier =
+        ImmutableMap.of("tracestate", "b3=4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-1");
+
+    // when
+    final SpanContext result = format.extract(carrier, getter);
+
+    // then
+    assertEquals("4bf92f3577b34da6a3ce929d0e0e4736", result.getTraceId().toLowerBase16());
+    assertEquals("00f067aa0ba902b7", result.getSpanId().toLowerBase16());
+    assertTrue(result.getTraceOptions().isSampled());
+  }
+
+  @Test
+  public final void verifyExtractParsesNotYetSampledRootSpanFromB3()
+      throws SpanContextParseException {
+    // given
+    final Map<String, String> carrier =
+        ImmutableMap.of("b3", "4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7");
+
+    // when
+    final SpanContext result = format.extract(carrier, getter);
+
+    // then
+    assertEquals("4bf92f3577b34da6a3ce929d0e0e4736", result.getTraceId().toLowerBase16());
+    assertEquals("00f067aa0ba902b7", result.getSpanId().toLowerBase16());
+    assertFalse(result.getTraceOptions().isSampled());
+  }
+
+  @Test
+  public final void verifyExtractParsesNotYetSampledRootSpanFromTracestate()
+      throws SpanContextParseException {
+    // given
+    final Map<String, String> carrier =
+        ImmutableMap.of("tracestate", "b3=4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7");
+
+    // when
+    final SpanContext result = format.extract(carrier, getter);
+
+    // then
+    assertEquals("4bf92f3577b34da6a3ce929d0e0e4736", result.getTraceId().toLowerBase16());
+    assertEquals("00f067aa0ba902b7", result.getSpanId().toLowerBase16());
+    assertFalse(result.getTraceOptions().isSampled());
+  }
+
+  @Test
+  public final void verifyExtractParsesDebugSpanFromB3() throws SpanContextParseException {
+    // given
+    final Map<String, String> carrier =
+        ImmutableMap.of(
+            "b3", "80f198ee56343ba864fe8b2a57d3eff7-e457b5a2e4d86bd1-d-05e3ac9a4f6e3b90");
+
+    // when
+    final SpanContext result = format.extract(carrier, getter);
+
+    // then
+    assertEquals("80f198ee56343ba864fe8b2a57d3eff7", result.getTraceId().toLowerBase16());
+    assertEquals("e457b5a2e4d86bd1", result.getSpanId().toLowerBase16());
+    assertFalse(result.getTraceOptions().isSampled());
```

Comment:
I'm not sure if this is the expected behaviour. I also was not sure of the appropriate way to propagate the "debug" flag. I would appreciate if someone could clarify.

Reply:
debug was dropped from w3c spec. so in this case it is lost. In normal zipkin instrumentation, we propagate it and also set span.debug=true when recording